quarkchain/tools/check_syncing_state.py

```diff
@@ -0,0 +1,79 @@
+#! /usr/bin/env python3
+
+import argparse
+import logging
+import time
+from datetime import datetime
+import jsonrpcclient
+import psutil
+import numpy
+from decimal import Decimal
+
+
+# disable jsonrpcclient verbose logging
+logging.getLogger("jsonrpcclient.client.request").setLevel(logging.WARNING)
+logging.getLogger("jsonrpcclient.client.response").setLevel(logging.WARNING)
+
+
+def now():
+    return datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+def checkHeight(private_client, public_client):
+    s_p = private_client.send(jsonrpcclient.Request("getRootBlockByHeight"))
```

Comment:
what happens when this throws an exception? it looks to me the whole program will halt

Reply:
I have added the timeout and exception here.